tokio-macros/src/lib.rs

```diff
@@ -14,6 +14,9 @@
 
 //! Macros for use with Tokio
 
+// This `extern` is required for older `rustc` versions but newer `rustc`
+// versions warn about the unused `extern crate`.
+#[allow(unused_extern_crates)]
```

Comment:
I'm not 100% sure if this is the best strategy, but the `extern crate proc_macro` is required in Rust 1.40 but not on nightly.

Reply:
This looks like the most reasonable way around it. For reference, it was changed in `cargo` in https://github.com/rust-lang/cargo/pull/7700.